environments/common/src/main/java/org/jboss/weld/environment/util/Reflections.java

```diff
@@ -123,8 +132,8 @@ public static boolean isClassLoadable(ResourceLoader resourceLoader, String clas
         }
         try {
             return cast(Class.forName(className));
-        } catch (Exception e) {
-            throw CommonLogger.LOG.cannotLoadClass(className, e);
+        } catch (Exception | LinkageError e) {
+            throw new ResourceLoadingException("Error loading class " + className, e);
```

Comment:
I think we should use `CommonLogger.LOG.cannotLoadClass()` here. `org.jboss.weld.resources.spi.ResourceLoader` is related to `ResourceLoader`. Also the `cannotLoadClass()` method is only used here...